monitoring/timedrift_test.go

```diff
@@ -0,0 +1,189 @@
+
+func (s *TimeDriftSuite) SetUpSuite(c *check.C) {
+	s.c = &timeDriftChecker{self: node1}
+	s.clock = clockwork.NewFakeClock()
+}
+
+func (s *TimeDriftSuite) TestTimeDriftChecker(c *check.C) {
+	tests := []struct {
+		Comment string
+		Nodes   []serf.Member
+		Times   map[string]time.Time
+		Result  []*pb.Probe
+	}{
+		{
+			Comment: "Time drift to both nodes is acceptable",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(100 * time.Millisecond),
+				node3.Name: s.clock.Now().Add(200 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.successProbe(node2, 100*time.Millisecond),
+				s.c.successProbe(node3, 200*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to both nodes is acceptable, one node is lagging behind",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(100 * time.Millisecond),
+				node3.Name: s.clock.Now().Add(-100 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.successProbe(node2, 100*time.Millisecond),
+				s.c.successProbe(node3, -100*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to node-3 is high",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(100 * time.Millisecond),
+				node3.Name: s.clock.Now().Add(500 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.successProbe(node2, 100*time.Millisecond),
+				s.c.failureProbe(node3, 500*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to node-2 is high",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(-time.Second),
+				node3.Name: s.clock.Now().Add(100 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.failureProbe(node2, -time.Second),
+				s.c.successProbe(node3, 100*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to both nodes is high",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(-time.Second),
+				node3.Name: s.clock.Now().Add(time.Second),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.failureProbe(node2, -time.Second),
+				s.c.failureProbe(node3, time.Second),
+			},
+		},
+	}
+	for _, test := range tests {
+		clients, err := ttlmap.New(10)
```

Comment:
Why `10`? If it's irrelevant then it only confuses the test, hide this behind an initialization function:
```go
clients := newClients(c)
// or
clients := newClients(c, test.Times, nodes) // if it depends on the test cases
```